src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/PossibleDeNovo.java

```diff
@@ -38,48 +40,67 @@
  *
  */
 @DocumentedFeature(groupName=HelpConstants.DOC_CAT_ANNOTATORS, groupSummary=HelpConstants.DOC_CAT_ANNOTATORS_SUMMARY, summary="Existence of a de novo mutation in at least one of the given families (hiConfDeNovo, loConfDeNovo)")
-public final class PossibleDeNovo extends InfoFieldAnnotation {
+public final class PossibleDeNovo extends PedigreeAnnotation {
+    private final MendelianViolation mendelianViolation;
+    private Set<Trio> trios;
 
+    @VisibleForTesting
     public PossibleDeNovo(final Set<Trio> trios, final double minGenotypeQualityP) {
+        super((Set<String>) null);
         this.trios = Collections.unmodifiableSet(new LinkedHashSet<>(trios));
         mendelianViolation = new MendelianViolation(minGenotypeQualityP);
     }
 
-    /**
-     * This is dummy constructor that will do nothing until https://github.com/broadinstitute/gatk/issues/1880 is addressed
-     */
+    public PossibleDeNovo(final Set<String> founderIds){
+        super(founderIds);
+        mendelianViolation = new MendelianViolation(DEFAULT_MIN_GENOTYPE_QUALITY_P);
+    }
+
+    public PossibleDeNovo(final File pedigreeFile){
+        super(pedigreeFile);
+        mendelianViolation = new MendelianViolation(DEFAULT_MIN_GENOTYPE_QUALITY_P);
+    }
+
     public PossibleDeNovo(){
-        this(Collections.emptySet(), 0);
+        this((Set<String>) null);
     }
 
     protected final OneShotLogger warning = new OneShotLogger(this.getClass());
 
-
+    // Static thresholds for the denovo calculation
+    public final static double DEFAULT_MIN_GENOTYPE_QUALITY_P = 0; // TODO should this be exposed as a command line argument?
```

Comment:
Is this possible in Barclay already?  I remember we planned to make that a feature.

Reply:
It is indeed possible now. You can add arguments to annotations just like you can for ReadFilters. Should this be exposed an argument? I'm not sure if it ever was in GATK3 or if it would mean much.